src/org/zaproxy/zap/extension/pscan/PassiveScanThread.java

```diff
@@ -141,7 +147,9 @@ public void run() {
 								if (shutDown) {
 									return;
 								}
-								if (scanner.isEnabled() && scanner.appliesToHistoryType(href.getHistoryType())) {
+								int hrefHistoryType = href.getHistoryType();
+								if (scanner.isEnabled() && (scanner.appliesToHistoryType(hrefHistoryType)
+										|| getOptedInHistoryTypes().contains(hrefHistoryType))) {
```

Comment:
This could use `optedInHistoryTypes` directly, to avoid the `Collections` wrapping.